src/components/engagement/engagement.service.ts

```diff
@@ -620,4 +623,9 @@ export class EngagementService {
       );
     }
   }
+
+  async privilegesFor(id: ID, session: Session) {
+    const engagement = await this.repo.readOne(id, session);
+    return this.privileges.for(session, IEngagement, engagement);
+  }
```

Comment:
I'm not sold on this. It's clunky.
The only reason this is here is to access `repo.readOne`, which gives us an unsecured version of the object to provide to the privileges.

This thing is called "object" in lots of places, but I've started transitioning to call it "context". The actual shape of it is very unclear. Its only purpose is to provide info to conditions. 

Currently these conditions don't use any secured info, like `scope`, `sensitivity`.
There's potential for usage like `Project.step` condition, which is secured. So if the condition needed it but the user couldn't read it then what happens. Admittedly this is a super edge case. 
And could be handled like user cannot read, then condition is false regardless of value. That would probably be the easiest, and I don't think we limit read access to this prop anyways.
Another option would be to provide the actual value internally even when publicly the user has no read access.

There's also more and more situations where the context needs to be customized for edge cases, using the `with*` helpers. I think that needs some refinement as well.

So the need for this context to be the unsecured object is questionable and I'm not even sure it should be typed that strictly.

---

So we could do something like
```ts
const engagement = await ResourceLoader.load(IEngagement, id);
const privileges = Privileges.for(session, IEngagement, engagement);
```
Which is already setup and doesn't need another boilerplate method.
(Given the type is adjusted in `Privileges`).